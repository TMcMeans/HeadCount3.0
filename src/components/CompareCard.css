@import url('https://fonts.googleapis.com/css?family=Poppins');

.red {
  color: grey;
}

.red:before {
  content: '- ';
}

.green {
  color: #4a314d;
  /* font-weight: bold; */
}

.green:before {
  content: '+ ';
}

.compare-card {
  border: 1px solid #f0af4d;
  border-radius: 4px;
  padding: 0.5rem;
  font-family: 'Poppins', sans-serif;
  border-radius: 0px;
  text-align: center;
  background: white;
  cursor: pointer;
}

.compare-card:hover {
  background: orange;
}

.compare-card ul {
  list-style: none;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-inline-start: 0px;
  font-weight: bold;
}

h1 {
  font-size: 1.2em;
}
