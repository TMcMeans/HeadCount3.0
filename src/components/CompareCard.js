import React from 'react';
import PropTypes from 'prop-types';

export const CompareCard = ({ location, stats, removeCompareSelections }) => {
  const data = Object.entries(stats);

  const dataGroup = data.map(datapoint => (
    <li className={datapoint[1] > 0.5 ? 'green' : 'red'} key={Math.random()}>
      {datapoint[0]}: {datapoint[1]}
    </li>
  ));
  return (
    <div
      className="compare-card"
      onClick={() => {
        removeCompareSelections(location);
      }}
    >
      <h1>{location}</h1>
      <ul>{dataGroup}</ul>
    </div>
  );
};
