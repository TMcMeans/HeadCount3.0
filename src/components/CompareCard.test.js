import React from 'react';
import { shallow } from 'enzyme';

import { CompareCard } from './CompareCard';

describe('Compare Card', () => {
  it('should match snapshot with all html elements rendered with correct data', () => {
    const mockFunc = jest.fn();
    const mockLocation = 'COLORADO';
    const mockStats = { 2012: 1, 2013: 1, 2014: 0, 2015: 0.552 };
    const wrapper = shallow(
      <CompareCard
        stats={mockStats}
        location={mockLocation}
        removeCompareSelections={mockFunc}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
