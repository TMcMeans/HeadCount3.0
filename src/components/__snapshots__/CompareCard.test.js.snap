// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compare Card should match snapshot with all html elements rendered with correct data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    location="COLORADO"
    removeCompareSelections={[Function]}
    stats={
        Object {
            "2012": 1,
            "2013": 1,
            "2014": 0,
            "2015": 0.552,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          COLORADO
</h1>,
        <ul>
          <li
                    className="green"
          >
                    2012
                    : 
                    1
          </li>
          <li
                    className="green"
          >
                    2013
                    : 
                    1
          </li>
          <li
                    className="red"
          >
                    2014
                    : 
                    0
          </li>
          <li
                    className="green"
          >
                    2015
                    : 
                    0.552
          </li>
</ul>,
      ],
      "className": "compare-card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "COLORADO",
        },
        "ref": null,
        "rendered": "COLORADO",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="green"
>
              2012
              : 
              1
</li>,
            <li
              className="green"
>
              2013
              : 
              1
</li>,
            <li
              className="red"
>
              2014
              : 
              0
</li>,
            <li
              className="green"
>
              2015
              : 
              0.552
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0.023555775383872213",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2012",
                ": ",
                1,
              ],
              "className": "green",
            },
            "ref": null,
            "rendered": Array [
              "2012",
              ": ",
              1,
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "0.6617920946848523",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2013",
                ": ",
                1,
              ],
              "className": "green",
            },
            "ref": null,
            "rendered": Array [
              "2013",
              ": ",
              1,
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "0.7496505496444283",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2014",
                ": ",
                0,
              ],
              "className": "red",
            },
            "ref": null,
            "rendered": Array [
              "2014",
              ": ",
              0,
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "0.17947100248360015",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2015",
                ": ",
                0.552,
              ],
              "className": "green",
            },
            "ref": null,
            "rendered": Array [
              "2015",
              ": ",
              0.552,
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            COLORADO
</h1>,
          <ul>
            <li
                        className="green"
            >
                        2012
                        : 
                        1
            </li>
            <li
                        className="green"
            >
                        2013
                        : 
                        1
            </li>
            <li
                        className="red"
            >
                        2014
                        : 
                        0
            </li>
            <li
                        className="green"
            >
                        2015
                        : 
                        0.552
            </li>
</ul>,
        ],
        "className": "compare-card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "COLORADO",
          },
          "ref": null,
          "rendered": "COLORADO",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="green"
>
                2012
                : 
                1
</li>,
              <li
                className="green"
>
                2013
                : 
                1
</li>,
              <li
                className="red"
>
                2014
                : 
                0
</li>,
              <li
                className="green"
>
                2015
                : 
                0.552
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0.023555775383872213",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2012",
                  ": ",
                  1,
                ],
                "className": "green",
              },
              "ref": null,
              "rendered": Array [
                "2012",
                ": ",
                1,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "0.6617920946848523",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2013",
                  ": ",
                  1,
                ],
                "className": "green",
              },
              "ref": null,
              "rendered": Array [
                "2013",
                ": ",
                1,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "0.7496505496444283",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2014",
                  ": ",
                  0,
                ],
                "className": "red",
              },
              "ref": null,
              "rendered": Array [
                "2014",
                ": ",
                0,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "0.17947100248360015",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2015",
                  ": ",
                  0.552,
                ],
                "className": "green",
              },
              "ref": null,
              "rendered": Array [
                "2015",
                ": ",
                0.552,
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
